upcloud/service/account.go

```diff
@@ -0,0 +1,105 @@
+package service
+
+import (
+	"encoding/json"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+)
+
+type Account interface {
+	GetAccountList() (upcloud.AccountList, error)
+	GetAccount() (*upcloud.Account, error)
+	GetAccountDetails(r *request.GetAccountDetailsRequest) (*upcloud.AccountDetails, error)
+	CreateSubaccount(r *request.CreateSubaccountRequest) (*upcloud.AccountDetails, error)
+	ModifySubaccount(r *request.ModifySubaccountRequest) error
```

Comment:
Pity we can't add the updated subaccount to the return values here ;/

Reply:
@Darep what do you think? All the other `service.Modify*` functions return object and error.  We could change `ModifySubaccount` to return also `(*upcloud.AccountDetails, error)` but that would mean making two requests (POST and then GET) in `ModifySubaccount` . Personally I would change the interface and make two requests and also make interface consistent with other interfaces.

Reply:
Let's make it consistent :blush: So, one more GET 😄